### ISSUE TITLE
Utility that exposes HTML/CSS/JS for the use of creating in-game browser tools

### ISSUE THREAD

**Issue by Fair-Tex** [Yellow-Dog-Man/Resonite-Issues#485]
### Is your feature request related to a problem? Please describe.

Currently there is nothing exposed that understands HTML/CSS/JS which means it's not possible to create an item such as a web browser

### Describe the solution you'd like

New nodes that would allow for the creation of a web browser. I think a spawnable shareable web browser would be hugely beneficial and could be a super fun tool that would keep people in reasonite longer. This would allow you to browse the internet with friends and more easily enjoy online content together in a VR setting

### Describe alternatives you've considered

not sure what other alternatives might exist for this

**Comment by Psychpsyo** [comment]
"not possible" is a strong wording...
But I agree that an in-game web browser would be nice, though I'm not sure if implementing its features as nodes would make sense,
I'd lean on the side of having a WebBrowser component that provides video and audio streams and can be fed a URL or html document to load.

**Comment by Fair-Tex** [comment]
> "not possible" is a strong wording... But I agree that an in-game web browser would be nice, though I'm not sure if implementing its features as nodes would make sense, I'd lean on the side of having a WebBrowser component that provides video and audio streams and can be fed a URL or html document to load.

Yeah, I wasn't sure if nodes would be the way to do it...not savvy enough with Resonite to know how this would be implemented, but I do think it would be an amazing feature no matter the implementation. This would be my #1 ask aside from performance related things.

**Comment by Frooxius** [comment]
This is a bit of a tricky issue. I feel there's two main things:

1) In-game web browser - this is something that could *potentially* be added, as a mostly "black box" tool. We'd have to evaluate this, but on itself, that's possible

2) Nodes to create your own web browser - if you want to build your own web browser with ProtoFlux nodes... this is HUGE undertaking. Like probably dozens of people working for years to get something that supports a good chunk of web out there and probably not something you'd want to do. 

I'm not sure what exactly is being asked for there. If it's nodes for parsing processing HTML/CSS/JS? That's a huge undertaking and if you want those, then we're likely just to say no to that request.

If you want just a web browser "view", regardless of how it's implemented, then it's something we can consider. But first I'd like to understand what you're actually interested in the first place.
